### ISSUE TITLE
"Error grabbing Grape" when using Neo4j 3.1.1 in Groovy script

### ISSUE THREAD

**Issue by h3xstream** [neo4j/neo4j#8739]
The dependencies can not be imported properly in a Groovy script using `@Grab` annotations. **org.ow2.asm : asm :  5.0.4** is not resolve.
The same script did work with the version 3.0.8 of neo4j Java API (previous version).

**Neo4j Version:** 3.1.1
**Operating System:** Windows 10
**API:** Java API

### Steps to reproduce

1. Run the script : `TestGraphApi.groovy`

```groovy
@Grab(group='org.neo4j', module='neo4j', version='3.1.1')
import org.neo4j.graphdb.GraphDatabaseFactory;


graphDb = new GraphDatabaseFactory().newEmbeddedDatabase( "hello.db" );
println "Database started"
```

### Expected behavior

The script should be able to run successfully.

### Actual behavior
```
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
General error during conversion: Error grabbing Grapes -- [download failed: org.ow2.asm#asm;5.0.4!asm.jar]

java.lang.RuntimeException: Error grabbing Grapes -- [download failed: org.ow2.asm#asm;5.0.4!asm.jar]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:83)
        at org.codehaus.groovy.reflection.CachedConstructor.doConstructorInvoke(CachedConstructor.java:77)
        at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrap.callConstructor(ConstructorSite.java:84)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:60)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:235)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:247)
        at groovy.grape.GrapeIvy.getDependencies(GrapeIvy.groovy:422)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSite.invoke(PogoMetaMethodSite.java:169)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:59)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:52)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:154)
        at groovy.grape.GrapeIvy.resolve(GrapeIvy.groovy:569)
        at groovy.grape.GrapeIvy$resolve$1.callCurrent(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:52)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:154)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:190)
        at groovy.grape.GrapeIvy.resolve(GrapeIvy.groovy:536)
        at groovy.grape.GrapeIvy$resolve$0.callCurrent(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:52)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:154)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:182)
        at groovy.grape.GrapeIvy.grab(GrapeIvy.groovy:258)
        at groovy.grape.Grape.grab(Grape.java:167)
        at groovy.grape.GrabAnnotationTransformation.visit(GrabAnnotationTransformation.java:378)
        at org.codehaus.groovy.transform.ASTTransformationVisitor$3.call(ASTTransformationVisitor.java:321)
        at org.codehaus.groovy.control.CompilationUnit.applyToSourceUnits(CompilationUnit.java:931)
        at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:593)
        at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:569)
        at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:546)
        at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:298)
        at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:268)
        at groovy.lang.GroovyShell.parseClass(GroovyShell.java:688)
        at groovy.lang.GroovyShell.run(GroovyShell.java:517)
        at groovy.lang.GroovyShell.run(GroovyShell.java:507)
        at groovy.ui.GroovyMain.processOnce(GroovyMain.java:652)
        at groovy.ui.GroovyMain.run(GroovyMain.java:384)
        at groovy.ui.GroovyMain.process(GroovyMain.java:370)
        at groovy.ui.GroovyMain.processArgs(GroovyMain.java:129)
        at groovy.ui.GroovyMain.main(GroovyMain.java:109)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.tools.GroovyStarter.rootLoader(GroovyStarter.java:109)
        at org.codehaus.groovy.tools.GroovyStarter.main(GroovyStarter.java:131)

1 error
```

```
$ grape resolve org.neo4j neo4j 3.1.1
Error in resolve:
        Error grabbing Grapes -- [download failed: org.ow2.asm#asm;5.0.4!asm.jar]
```

As a comparison Neo4j 3.0.8 which is working.
```
$ grape resolve org.neo4j neo4j 3.0.8
\.groovy\grapes\org.neo4j\neo4j\jars\neo4j-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-kernel\jars\neo4j-kernel-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-graphdb-api\jars\neo4j-graphdb-api-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-common\jars\neo4j-common-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-resource\jars\neo4j-resource-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-collections\jars\neo4j-collections-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-primitive-collections\jars\neo4j-primitive-collections-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-unsafe\jars\neo4j-unsafe-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-io\jars\neo4j-io-3.0.8.jar
\.groovy\grapes\org.apache.commons\commons-lang3\jars\commons-lang3-3.3.2.jar
\.groovy\grapes\org.neo4j\neo4j-csv\jars\neo4j-csv-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-logging\jars\neo4j-logging-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-security\jars\neo4j-security-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-lucene-upgrade\jars\neo4j-lucene-upgrade-3.0.8.jar
\.groovy\grapes\org.apache.lucene\lucene-core\jars\lucene-core-5.5.0.jar
\.groovy\grapes\org.apache.lucene\lucene-backward-codecs\jars\lucene-backward-codecs-5.5.0.jar
\.groovy\grapes\org.neo4j\neo4j-lucene-index\jars\neo4j-lucene-index-3.0.8.jar
\.groovy\grapes\org.apache.lucene\lucene-analyzers-common\jars\lucene-analyzers-common-5.5.0.jar
\.groovy\grapes\org.apache.lucene\lucene-queryparser\jars\lucene-queryparser-5.5.0.jar
\.groovy\grapes\org.apache.lucene\lucene-codecs\jars\lucene-codecs-5.5.0.jar
\.groovy\grapes\org.neo4j\neo4j-graph-algo\jars\neo4j-graph-algo-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-udc\jars\neo4j-udc-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-graph-matching\jars\neo4j-graph-matching-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-cypher\jars\neo4j-cypher-3.0.8.jar
\.groovy\grapes\org.scala-lang\scala-library\jars\scala-library-2.11.8.jar
\.groovy\grapes\org.scala-lang\scala-reflect\jars\scala-reflect-2.11.8.jar
\.groovy\grapes\org.neo4j\neo4j-codegen\jars\neo4j-codegen-3.0.8.jar
\.groovy\grapes\org.ow2.asm\asm\jars\asm-5.0.2.jar
\.groovy\grapes\org.neo4j\neo4j-cypher-compiler-2.3\jars\neo4j-cypher-compiler-2.3-2.3.8.jar
\.groovy\grapes\org.neo4j\neo4j-cypher-frontend-2.3\jars\neo4j-cypher-frontend-2.3-2.3.8.jar
\.groovy\grapes\org.parboiled\parboiled-scala_2.11\jars\parboiled-scala_2.11-1.1.7.jar
\.groovy\grapes\org.parboiled\parboiled-core\jars\parboiled-core-1.1.7.jar
\.groovy\grapes\com.googlecode.concurrentlinkedhashmap\concurrentlinkedhashmap-lru\jars\concurrentlinkedhashmap-lru-1.4.2.jar
\.groovy\grapes\org.neo4j\neo4j-cypher-compiler-3.0\jars\neo4j-cypher-compiler-3.0-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-cypher-frontend-3.0\jars\neo4j-cypher-frontend-3.0-3.0.8.jar
\.groovy\grapes\net.sf.opencsv\opencsv\jars\opencsv-2.3.jar
\.groovy\grapes\org.neo4j\neo4j-jmx\jars\neo4j-jmx-3.0.8.jar
\.groovy\grapes\org.neo4j\neo4j-consistency-check\jars\neo4j-consistency-check-3.0.8.jar
```

### Temporary workaround

```groovy
@Grab(group='org.neo4j', module='neo4j', version='3.0.8')
```

**Comment by chrisvest** [comment]
The dependency does exist on Maven Central, though: http://search.maven.org/#artifactdetails%7Corg.ow2.asm%7Casm%7C5.0.4%7Cjar

So perhaps the problem is either temporary, or exclusive to your environment?

**Comment by h3xstream** [comment]
I will investigate more .. because it was working with Gradle. Perhaps the repo URL was different..

**Comment by h3xstream** [comment]
**Got it! This is unrelated to Neo4J.**

I had a partially download the asm dependency in my maven cache *".m2\repository\org\ow2\asm\asm\5.0.4"* where no jar was present. The Grape module is looking at the Maven cache prior downloading new jars. It create a reference to the .m2 directory to avoid downloading it again.

Solution: I deleted the folder 5.0.4 (.m2\repository\org\ow2\asm\asm\5.0.4) and I was able to run the script. The rest of the dependencies were also downloaded transparently by Grape.
